packages/binary/messagelayer/messageparser/events.go

```diff
@@ -11,3 +15,41 @@ type Events struct {
 	// Fired when a message got rejected by a filter.
 	MessageRejected *events.Event
 }
+
+// MessageParsedEvent represents the parameters of messageParsedEvent
+type MessageParsedEvent struct {
+	Message *message.Message
+	Peer    *peer.Peer
+}
+
+// BytesRejectedEvent represents the parameters of bytesRejectedEvent
+type BytesRejectedEvent struct {
+	Bytes []byte
+	Peer  *peer.Peer
+}
+
+// MessageRejectedEvent represents the parameters of messageRejectedEvent
+type MessageRejectedEvent struct {
+	Message *message.Message
+	Peer    *peer.Peer
+}
+
+func newEvents() *Events {
+	return &Events{
+		MessageParsed:   events.NewEvent(messageParsedEvent),
+		BytesRejected:   events.NewEvent(bytesRejectedEvent),
+		MessageRejected: events.NewEvent(messageRejectedEvent),
+	}
+}
+
+func messageParsedEvent(handler interface{}, params ...interface{}) {
+	handler.(func(*MessageParsedEvent))(params[0].(*MessageParsedEvent))
+}
+
+func bytesRejectedEvent(handler interface{}, params ...interface{}) {
+	handler.(func(*BytesRejectedEvent, error))(params[0].(*BytesRejectedEvent), params[1].(error))
```

Comment:
Just wondering about the design choice to not include the error in the event. Is there a reason to not do it? 
I've seen it before as well. Whatever the decision is it needs to be consistent.

Reply:
I just have a feeling that the error shouldn't be the member of the struct, it should be the external checkpoint before we access the data fields.